### PR TITLE
Feat/crawl erc20 contract

### PR DESCRIPTION
**Pull request by phamphong9981**
## Summary by CodeRabbit

- **New Features**
	- Introduced configuration for ERC20 smart contracts including limits, timing, and job repetition settings.
	- Added polyfill support for the `fetch` API to ensure compatibility across different environments.
	- Implemented database schema for storing ERC20 contract information.
	- Enhanced system capabilities to handle ERC20 contracts, including job scheduling and batch processing.
	- Developed a service class for interacting with ERC20 contracts, supporting updates and Ethereum smart contract interactions.

### CODE REVIEW

**Comment by kienvc** [comment]
@coderabbitai review

File: fetch-polyfill.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/* eslint-disable */
+// fetch-polyfill.js
```
Comment (peara) [inline fetch-polyfill.js:2]: required by viem: https://viem.sh/docs/compatibility
will be removed after upgrading node to v18

File: config.json [hunk 2]
```diff
@@ -383,5 +383,11 @@
     "millisecondCrawl": 1000,
     "recordsPerCall": 100,
     "solcRepo": null
+  },
+  "erc20": {
+    "limitEvmSmartContractGet": 50,
```
Comment (peara) [inline config.json:388]: thừa này

File: src/common/utils/etherjs_client.ts [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
-  constructor() {
+  constructor(batch?: boolean) {
     const selectedChain = networks.find(
       (network) => network.chainId === config.chainId
     );
     if (selectedChain?.EVMJSONRPC) {
-      this.etherJsClient = ethers.getDefaultProvider(
-        selectedChain.EVMJSONRPC[0]
-      );
+      if (batch) {
```
Comment (peara) [inline src/common/utils/etherjs_client.ts:14]: bỏ đoạn này đi, để nguyên nó như cũ

File: src/common/utils/etherjs_client.ts [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+      }
     } else {
       throw new Error(`EVMJSONRPC not found with chainId: ${config.chainId}`);
     }
   }
+
+  public static getViemClient(): PublicClient {
+    const selectedChain = networks.find(
+      (network) => network.chainId === config.chainId
+    );
```
Comment (peara) [inline src/common/utils/etherjs_client.ts:31]: if (!selectedChain.EVMJSONRPC) throw ...

**Pull request merged**